packages/components/inputs/rich-text-utils/src/html/html.tsx

```diff
@@ -1,186 +1,251 @@
 
-  // Add a new rule that handles marks...
-  {
-    deserialize(el: Element, next: (arg0: NodeListOf<ChildNode>) => void) {
-      const type =
-        MARK_TAGS[el.tagName.toLowerCase() as keyof typeof MARK_TAGS];
-      if (type) {
-        return {
-          object: 'mark',
-          type,
-          nodes: next(el.childNodes),
-        };
-      }
-      return;
-    },
-    serialize(obj: TSerializableObject, children: ReactNode) {
-      if (obj.object === 'mark') {
-        switch (obj.type) {
-          case 'span':
-            return <span>{children}</span>;
-          case 'bold':
-            return <strong>{children}</strong>;
-          case 'italic':
-            return <em>{children}</em>;
-          case 'underlined':
-            return <u>{children}</u>;
-          case 'superscript':
-            return <sup>{children}</sup>;
-          case 'subscript':
-            return <sub>{children}</sub>;
-          case 'strikethrough':
-            return <del>{children}</del>;
-          case 'code':
-            return (
-              <pre>
-                <code>{children}</code>
-              </pre>
-            );
-        }
+const deserializeElement = (
+  el: HTMLElement | ChildNode
+): Deserialized | Deserialized[] => {
+  // https://developer.mozilla.org/en-US/docs/Web/API/Node/nodeType#value
+  if (el.nodeType === 3) {
+    return wrapWithParagraphIfRootElement(el, { text: el.textContent || '' }); // for root TEXT_NODE -> wrap with <p>
+  } else if (el.nodeType !== 1) {
+    return null; // for non-ELEMENT_NODE
+  }
+
+  const { nodeName } = el;
+  let parent = el;
+
+  if (
+    nodeName === 'PRE' &&
+    el.childNodes[0] &&
+    el.childNodes[0].nodeName === 'CODE'
+  ) {
+    parent = el.childNodes[0];
+  }
+  let children: Deserialized[] = Array.from(parent.childNodes)
+    .map(deserializeElement)
+    .flat();
+
+  if (children.length === 0) {
+    children = [{ text: '' }];
+  }
+
+  if (el.nodeName === 'BODY') {
+    return jsx('fragment', {}, children);
+  }
+
+  if (el.nodeName === 'SPAN') {
+    let attrs = {};
+    const styleStr = (el as HTMLElement).getAttribute('style');
+    const styleObj = parse(styleStr || '');
+
+    if (isEmpty(styleObj)) {
+      // if no style attrs -> just use `span`
+      return wrapWithParagraphIfRootElement(
+        el,
+        jsx('element', { type: 'span' }, children)
+      );
+    } else {
+      attrs = Object.entries(styleObj || {}).reduce(
+        (mappedAttrObj, [key, value]) => {
+          const values = value.split(' '); // to cover the case of space separated values e.g. `text-decoration-line: "underline line-through"`
+
+          values.forEach((splittedValue) => {
+            if (mapper[key] && mapper[key][splittedValue]) {
+              // checking if the parsed style attr value has representation in the mapper obj
+              mappedAttrObj = {
+                ...mapper[key][splittedValue],
+                ...mappedAttrObj,
```

Comment:
Shouldn't this be the other way around? First spread, then overwrite.

```suggestion
                ...mappedAttrObj,
                ...mapper[key][splittedValue],
```